template/template_test.go

```diff
@@ -371,10 +381,19 @@ func checkVersionBody(t *testing.T, body string, custom map[string]interface{})
 	assert.True(t, strings.Contains(body, toString(custom["repo_url"])))
 	assert.True(t, strings.Contains(body, toString(custom["scanned_at"])))
 
-	depArr := toArrMap(custom["version_updates"])
-	assert.NotNil(t, depArr)
+	dirDepArr := toArrMap(custom["direct_updates"])
+	assert.NotNil(t, dirDepArr)
 
-	for _, dep := range depArr {
+	transDepArr := toArrMap(custom["transitive_updates"])
+	assert.NotNil(t, transDepArr)
+
+	for _, dep := range dirDepArr {
+		assert.True(t, strings.Contains(body, toString(dep["name"])))
+		assert.True(t, strings.Contains(body, toString(dep["version"])))
+		assert.True(t, strings.Contains(body, toString(dep["latest_version"])))
+		assert.True(t, strings.Contains(body, toString(dep["ecosystem"])))
+	}
```

Comment:
this validation check look can be re-used. extract to some func and use it for both `direct_updates` and `transitive_updates`